.github/workflows/rust.yml

```diff
@@ -32,6 +32,28 @@ jobs:
       - name: Run tests
         run: RUST_LOG=debug bash contrib/test.sh
 
+  Test-Stable-macOS:
+    name: Stable tests on macOS
+    runs-on: macos-13
+    env:
+      # Override the value from the rust-toolchain file
+      RUSTUP_TOOLCHAIN: stable
+    steps:
+      - name: "Checkout repo"
+        uses: actions/checkout@v4
+      - name: "Install stable toolchain"
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: stable
+      - name: "Use cache"
+        uses: Swatinem/rust-cache@v2
+      - name: Setup Docker on macOS
+        uses: douglascamata/setup-docker-macos-action@v1.0.0
```

Comment:
It looks like you didn't remove this when using macos-latest, try again with this docker setup removed